api/src/libs/messaging.ts

```diff
@@ -46,10 +45,55 @@ export const contactsForProfile = async (profileId: number): Promise<string[]> =
   }
 }
 
+export const profileDetails = async (profileId: number): Promise<string> => {
+
+  try {
+    const dm = new DataManager(shared.pgPool);
+    const { ProfileModel } = dm;
+    const profiles = await ProfileModel.findById(profileId);
+    const profileName = profiles.name;
+    return profileName;
+  } catch (err) {
+    const message = `Unable to fetch profile ${profileId}`;
+    logger.error(`${message}, err = ${err.message}`);
+
+    return '';
+  }
+}
+
+export const updateEmailContent = async (buff: string, to: string[], profileName: string, contactNames: string[]): Promise<string> => {
+  try {
+    let emailContent: string;
+
+    const mapObj = {
+      POName: contactNames[0],
+      TCName: contactNames[1],
+      POEmail: to[0],
+      TCEmail: to[1],
+      projectName: profileName,
+    };
+
+    const re = new RegExp(Object.keys(mapObj).join('|'),'gi');
+    emailContent = buff.replace(re, matched => {
+      return mapObj[matched];
+    });
+    return emailContent;
+  } catch (err) {
+    const message = `Unable to update email content`;
+    logger.error(`${message}, err = ${err.message}`);
+
+    return '';
```

Comment:
It **may** be better to do a return type of `Promise<string | undefined>` and then just return rather than returning an empty string. That way you can check if the return value is `undefined` rather than an empty string.